src/Symfony/Component/Cache/Psr16Cache.php

```diff
@@ -19,6 +20,12 @@
 use Symfony\Component\Cache\Exception\InvalidArgumentException;
 use Symfony\Component\Cache\Traits\ProxyTrait;
 
+if (class_exists(InstalledVersions::class) && InstalledVersions::isInstalled('psr/simple-cache')) {
```

Comment:
We should use a feature test instead, based on reflection I suppose

Reply:
I've force-pushed 9a729120ba which checks for a return type and throws an exception if one is defined